addons/stock_account/views/stock_valuation_layer_views.xml

```diff
@@ -104,6 +139,9 @@
                 <field name="reference"/>
                 <field name="categ_id" />
                 <field name="product_tmpl_id" />
+                <field name="warehouse_id" groups="stock.group_stock_multi_warehouses"/>
```

Comment:
search on non-stored related fields won't work. You can either make it stored or write a search function for it.